collect_app/src/test/java/org/odk/collect/android/forms/FormsRepositoryTest.java

```diff
@@ -0,0 +1,40 @@
+package org.odk.collect.android.forms;
+
+import org.junit.Test;
+import org.odk.collect.android.utilities.FileUtils;
+
+import java.io.File;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+public abstract class FormsRepositoryTest {
+
+    public abstract FormsRepository buildSubject();
+
+    public abstract String getFormFilesPath();
+
+    @Test
+    public void get_whenFormHasNullVersion_returnsForm() {
```

Comment:
Given that the error here was the SQL query being constructed incorrectly it felt like the right time to start adding tests. Using a contract testing pattern like we're doing here lets us verify that our "real" (`DatabaseFormsRepository`) and "fake" (`InMemFormsRepository`) both pass the same set of tests. This lets us be more confident that we can switch the "real" object for the "fake" object in tests.

Reply:
I think another option would be to let the dao layer deal with queries and do testing at that level.